### PR TITLE
Add details on JSON

### LINKED ISSUE #829: [EDIT] General: JSON
### Type of Edit (select all that apply)

Add new content (definitions, codeblocks, etc.)

### Description (optional)

See [comment below](https://github.com/Codecademy/docs/issues/829#issuecomment-1187508337) for detailed description/instructions for this issue.

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by YanisaHS**
### Description

I added information to the following pages:
- General: JSON
  - Added ) after “EcmaScript” in the first paragraph
  - Noted that JavaScript can parse JSON directly
  - Noted that JavaScript is language independent with more details
- Python: JSON Module
  - Noted that JSON is based on JavaScript object-literal syntax
  - Added a list of data types that can be converted to JSON strings in Python
- JavaScript: JSON
  - Added info on how the JSON object is built into JavaScript

Closes #829 

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/json/json.md [hunk 1]
```diff
@@ -14,7 +14,9 @@ CatalogContent:
...
+**JavaScript Object Notation (JSON)** is a language-independent data format that is readable, writable, and parsable for both humans and machines. JSON is based on the syntax of the third edition of a [JavaScript](https://www.codecademy.com/resources/docs/javascript) standard known as ([ECMAScript](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/)). Many programming languages, such as [Python](https://www.codecademy.com/resources/docs/python), have implemented [libraries](https://www.codecademy.com/resources/docs/python/json-module) to parse and generate JSON-formatted data. JavaScript can parse JSON directly with the [`JSON` object](https://www.codecademy.com/resources/docs/javascript/json).
+
+Although JSON was originally based on a subset of JavaScript syntax, JSON is language independent and does not require JavaScript. JSON uses some conventions from C-descendent languages, such as C, C++, JavaScript, and Java.
```
Comment (SSwiniarski) [inline content/general/concepts/json/json.md:19]:
```suggestion
Although JSON was originally based on a subset of JavaScript syntax, JSON is language independent and does not require JavaScript. JSON uses some conventions from C-descendent languages, such as [C](https://www.codecademy.com/resources/docs/c), [C++](https://www.codecademy.com/resources/docs/cpp), JavaScript, and [Java](https://www.codecademy.com/resources/docs/java).
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@YanisaHS I completed the first review and only had one comment.

**Comment by YanisaHS** [comment]
Thanks @SSwiniarski! Just added those changes.

**Review by SSwiniarski — APPROVED** [review]
@YanisaHS Looks good for next review.

File: content/general/concepts/json/json.md [hunk 2]
```diff
@@ -14,7 +14,9 @@ CatalogContent:
...
+
+Although JSON was originally based on a subset of JavaScript syntax, JSON is language independent and does not require JavaScript. JSON uses some conventions from C-descendent languages, such as [C](https://www.codecademy.com/resources/docs/c), [C++](https://www.codecademy.com/resources/docs/cpp), JavaScript, and [Java](https://www.codecademy.com/resources/docs/java).
 
 JSON is commonly used for transferring data between clients and servers for tasks such as web browsing or form submission. Some companies also use JSON to allow their data to be accessed in other applications via [API](https://www.codecademy.com/resources/docs/general/api). Some examples include:
```
Comment (Dusch4593) [inline content/general/concepts/json/json.md:22]: On line # 50, can we link "Whitespace" to the following Docs entry?

https://www.codecademy.com/resources/docs/general/whitespace

Also, on this same line, can rewrite as this?

```
Whitespace (e.g., spaces, tabs, line feeds, and carriage returns) is ignored between names, values, and punctuation.
```

Comment (Dusch4593) [reply content/general/concepts/json/json.md:22]: Then on line # 52, let's link "comments" out to the following Docs entry: https://www.codecademy.com/resources/docs/general/comment

Comment (Dusch4593) [reply content/general/concepts/json/json.md:22]: Lastly for this file, on line # 81, let's link "XML" out to this Docs entry: https://www.codecademy.com/resources/docs/general/xml

File: content/python/concepts/json-module/json-module.md [hunk 3]
```diff
@@ -20,4 +20,15 @@ The `json` module provides tools for working with the [JavaScript Object Notatio
 import json
 ```
```
Comment (Dusch4593) [inline content/python/concepts/json-module/json-module.md:22]: On line # 17, **bold** the following: `json` module.

File: content/python/concepts/json-module/json-module.md [hunk 4]
```diff
@@ -20,4 +20,15 @@ The `json` module provides tools for working with the [JavaScript Object Notatio
...
 
-The following functions are provided by this module:
+JSON in Python is based on the object-literal syntax of JavaScript. The following Python [data types](https://www.codecademy.com/resources/docs/python/data-types) can be converted to JSON strings in Python:
```
Comment (Dusch4593) [inline content/python/concepts/json-module/json-module.md:23]:
```suggestion
JSON in Python is based on the object-literal syntax of [JavaScript](https://www.codecademy.com/resources/docs/javascript). The following Python [data types](https://www.codecademy.com/resources/docs/python/data-types) can be converted to JSON strings in Python:
```

File: content/python/concepts/json-module/json-module.md [hunk 5]
```diff
@@ -20,4 +20,15 @@ The `json` module provides tools for working with the [JavaScript Object Notatio
...
+
+- Tuple
+- List
+- Dict
+- String
+- Int
+- Float
+- Bool
+- NoneType
```
Comment (Dusch4593) [inline content/python/concepts/json-module/json-module.md:32]:
```suggestion
- Booleans (`bool` type)
- [Dictionaries](https://www.codecademy.com/resources/docs/python/dictionaries) (`dict` type)
- Number types (i.e., `int` and `float`)
- Sequence types (i.e., [lists](https://www.codecademy.com/resources/docs/python/lists) and [tuples](https://www.codecademy.com/resources/docs/python/tuples))
- [Strings](https://www.codecademy.com/resources/docs/python/strings) (`str` type)
- `NoneType`
```

File: content/javascript/concepts/json/json.md [hunk 6]
```diff
@@ -12,7 +12,7 @@ CatalogContent:
   - 'paths/full-stack-engineer-career-path'
 ---
 
-In JavaScript, the reserved `JSON` object is used to convert values to JavaScript Object Notation ([JSON](https://www.codecademy.com/resources/docs/general/json)). It can also be used for parsing JSON values.
+In JavaScript, the reserved `JSON` object is used to convert values to JavaScript Object Notation ([JSON](https://www.codecademy.com/resources/docs/general/json)). It can also be used for parsing JSON values. `JSON` is a standard object built directly into JavaScript, so there is no need to import any external packages. `JSON` can not be constructed, and there are no functions which can create `JSON` objects.
```
Comment (Dusch4593) [inline content/javascript/concepts/json/json.md:15]:
```suggestion
In JavaScript, the built-in **`JSON`** object is used to convert values to [JavaScript Object Notation (JSON)](https://www.codecademy.com/resources/docs/general/json) as well as parse those values. `JSON` is a standard JavaScript object, so there is no need to import any external packages.
```

I think the last sentence make more sense in the "Syntax" section.

File: content/javascript/concepts/json/json.md [hunk 7]
```diff
@@ -12,7 +12,7 @@ CatalogContent:
...
 
 ## Syntax
```
Comment (Dusch4593) [inline content/javascript/concepts/json/json.md:18]: Could we prepend the sentence on line # 23 with the following?

"No constructor function is needed to create `JSON` objects."

File: content/python/concepts/json-module/json-module.md [hunk 8]
```diff
@@ -20,4 +20,15 @@ The `json` module provides tools for working with the [JavaScript Object Notatio
 import json
```
Comment (Dusch4593) [inline content/python/concepts/json-module/json-module.md:20]: Above in the list of meta-`Tags` (lines # 7-11), let's add `'Modules'` to it

File: content/general/concepts/json/json.md [hunk 9]
```diff
@@ -14,7 +14,9 @@ CatalogContent:
...
+**JavaScript Object Notation (JSON)** is a language-independent data format that is readable, writable, and parsable for both humans and machines. JSON is based on the syntax of the third edition of a [JavaScript](https://www.codecademy.com/resources/docs/javascript) standard known as ([ECMAScript](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/)). Many programming languages, such as [Python](https://www.codecademy.com/resources/docs/python), have implemented [libraries](https://www.codecademy.com/resources/docs/python/json-module) to parse and generate JSON-formatted data. JavaScript can parse JSON directly with the [`JSON` object](https://www.codecademy.com/resources/docs/javascript/json).
+
+Although JSON was originally based on a subset of JavaScript syntax, JSON is language independent and does not require JavaScript. JSON uses some conventions from C-descendent languages, such as [C](https://www.codecademy.com/resources/docs/c), [C++](https://www.codecademy.com/resources/docs/cpp), JavaScript, and [Java](https://www.codecademy.com/resources/docs/java).
```
Comment (Dusch4593) [inline content/general/concepts/json/json.md:19]:
```suggestion
In addition to JavaScript, JSON uses some conventions from the [C](https://www.codecademy.com/resources/docs/c)-descendent languages, including C , [C++](https://www.codecademy.com/resources/docs/cpp), and [Java](https://www.codecademy.com/resources/docs/java).
```

I think the ideas from the first sentence is already covered in the previous paragraph.

File: content/general/concepts/json/json.md [hunk 10]
```diff
@@ -14,7 +14,9 @@ CatalogContent:
...
 
 JSON is commonly used for transferring data between clients and servers for tasks such as web browsing or form submission. Some companies also use JSON to allow their data to be accessed in other applications via [API](https://www.codecademy.com/resources/docs/general/api). Some examples include:
```
Comment (Dusch4593) [inline content/general/concepts/json/json.md:21]:
```suggestion
JSON is commonly used for transferring data between clients and [servers](https://www.codecademy.com/resources/docs/general/server) for tasks such as web browsing or form submission. Some companies also use JSON to allow their data to be accessed in other applications via [API](https://www.codecademy.com/resources/docs/general/api). Some examples include:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@YanisaHS Finished my review and left you some comments/suggestions 😄

**Comment by YanisaHS** [comment]
Thanks @Dusch4593 ! Just pushed the changes. 🥳

File: content/general/concepts/json/json.md [hunk 11]
```diff
@@ -14,9 +14,11 @@ CatalogContent:
```
Comment (Dusch4593) [inline content/general/concepts/json/json.md:14]: Could we add the following to the meta-`Tags` above?

- 'Properties'
- 'Values'

**Review by Dusch4593 — APPROVED** [review]
@YanisaHS LGTM! Just one last little suggestion.

**Comment by YanisaHS** [comment]
@Dusch4593 Yep! Done! 👼

**Comment by Dusch4593** [comment]
@YanisaHS This PR seems good for merge! 😄 I pushed a few last-minute changes in 567bb2b298c1c605f1af5eff3b3fcfb654dc3e45, including:

- Light edits to the rendered curly brackets `{ }` in General: JSON.
- In JavaScript: JSON, I embeded "JSON" in back-ticks within the meta-`Description`.

**Comment by Dusch4593** [comment]
Congrats @YanisaHS your changes should now be live! 🚀 🎊 🎉 Live links below.

- General: JSON - https://www.codecademy.com/resources/docs/general/json
- JavaScript: JSON - https://www.codecademy.com/resources/docs/javascript/json
- Python: JSON Module - https://www.codecademy.com/resources/docs/python/json-module

Let us know if everything looks ok on your end and if you can also see your profile credited on those entry pages. Looking forward to merging your next contribution! 😄

**Comment by YanisaHS** [comment]
@Dusch4593 Yay!! Looks amazing! 🥳🥳🎉

**Pull request merged**